src/main/java/org/apereo/cas/support/oauth/web/endpoints/OAuth20AuthorizeEndpointController.java

```diff
@@ -245,11 +245,14 @@ protected ModelAndView buildAuthorizationForRequest (
                 .map(String::valueOf)
                 .orElseGet(OAuth20GrantTypes.AUTHORIZATION_CODE::getType)
                 .toUpperCase();
-        val accessType = context
+        var accessType = context
                 .getRequestParameter(OsfCasOAuth20Constants.ACCESS_TYPE)
                 .map(String::valueOf)
-                .orElse(OsfCasOAuth20CodeType.ONLINE.name())
+                .orElse(StringUtils.EMPTY)
                 .toUpperCase();
+        if (!OsfCasOAuth20CodeType.OFFLINE.name().equalsIgnoreCase(accessType)) {
+            accessType = OsfCasOAuth20CodeType.ONLINE.name();
+        }
```

Comment:
Unlike other parameters in the authorization URL (e.g. `response_type`), the `access_type` is an OSF customization that doesn't have a validator. More specifically, invalid `response_type` is intercepted by the validator and an error page is returned; while invalid `access_type` is ignored and the default value is used. Thus, must add this extra check to apply the default value.